inversion-api/src/main/java/io/inversion/utils/RestClient.java

```diff
@@ -130,73 +134,85 @@
  * </pre>
  */
 public class RestClient {
-    static Log log = LogFactory.getLog(RestClient.class);
 
-    protected String name = null;
+    static Log                                       log              = LogFactory.getLog(RestClient.class);
+
+    protected String                                 name             = null;
 
     /**
      * Optional base url that will be prepended to the url arg of any calls assuming that the url arg supplied is a relative path and not an absolute url.
      */
-    protected String url = null;
+    protected String                                 url              = null;
 
     /**
      * Indicates the headers from the root inbound Request being handled on this Chain should be included on this request minus any blacklisted headers.
      */
-    protected boolean forwardHeaders = false;
+    protected boolean                                forwardHeaders   = false;
+
+    /**
+     * Indicates that a request body should be gzipped and the content-encoding header should be sent with value "gzip".
+     * <p>
+     * Valid values include "on", "off", "force" or the number of bytes of content that should trigger compression.
+     * <p>
+     * The default value is "1024" which implies "on" for anything over 1KB.
+     * <p>
+     * This property is modeled off of {@see org.apache.coyote.CompressionConfig.setCompression()}.
+     */
+    protected String                                 compression      = "1024";
 
     /**
      * Always forward these headers.
      *
      * @see #shouldForwardHeader(String)
      */
-    protected Set whitelistHeaders = new HashSet(Arrays.asList("authorization", "cookie", "x-forwarded-host", " x-forwarded-proto"));
+    protected Set                                    whitelistHeaders = new HashSet(Arrays.asList("authorization", "cookie", "x-forwarded-host", " x-forwarded-proto"));
 
     /**
      * Never forward these headers.
      *
      * @see #shouldForwardHeader(String)
      */
-    protected Set blacklistHeaders = new HashSet(Arrays.asList("content-length", "content-type", "content-encoding", "content-language", "content-location", "content-md5", "host"));
+    protected Set                                    blacklistHeaders = new HashSet(Arrays.asList("content-length", "content-type", "content-encoding", "content-language", "content-location", "content-md5", "host"));
```

Comment:
can we modify this name. It implies the restclient will not use these. But this really means that if an existing request has it you won't forward the header.

Reply:
instead of whitelistHeaders and blacklistHeaders, how about "forwardIncludeHeaders" and "forwardExcludeHeaders"?

Reply:
I like it.